extensions/amp-story/1.0/story-analytics.js

```diff
@@ -99,28 +108,74 @@ export class StoryAnalyticsService {
           return;
         }
 
-        this.triggerEvent(AnalyticsEvent.PAGE_VISIBLE);
+        this.triggerEvent(StoryAnalyticsEvent.PAGE_VISIBLE);
 
         const pageIds = this.storeService_.get(StateProperty.PAGE_IDS);
         const pageIndex = this.storeService_.get(
           StateProperty.CURRENT_PAGE_INDEX
         );
         if (pageIndex === pageIds.length - 1) {
-          this.triggerEvent(AnalyticsEvent.LAST_PAGE_VISIBLE);
+          this.triggerEvent(StoryAnalyticsEvent.LAST_PAGE_VISIBLE);
         }
       },
       true /* callToInitialize */
     );
   }
 
   /**
-   * @param {!AnalyticsEvent} eventType
+   * @param {!StoryAnalyticsEvent} eventType
    */
   triggerEvent(eventType) {
+    this.incrementEventCountForPage_(eventType);
     triggerAnalyticsEvent(
       this.element_,
       eventType,
-      this.variableService_.get()
+      this.getDetails_(eventType)
     );
   }
+
+  /**
+   * @visibleForTesting
+   * @return {!Object}
+   */
+  getVariableDetailsForTesting() {
+    return this.details_;
+  }
+
+  /**
+   * Gets details for a given event type.
+   * @param {!StoryAnalyticsEvent} eventType
+   * @private
+   * @return {!JsonObject}}
+   */
+  getDetails_(eventType) {
+    const details = {};
+    const vars = this.variableService_.get();
+    const pageId = vars['storyPageId'];
+
+    if (this.eventsPerPage_[pageId][eventType] > 1) {
+      Object.assign(details, {repeated: true});
+    }
+
+    this.details_ = /** @type {!JsonObject} */ (Object.assign(
+      {detailsForPage: details},
+      vars
+    ));
+    return this.details_;
+  }
+
+  /**
+   * Keeps count of number of events emitted by page for an event type.
+   * @param {!StoryAnalyticsEvent} eventType
+   * @private
+   */
+  incrementEventCountForPage_(eventType) {
```

Comment:
Nit: naming consistency between `incrementEventCountForPage` and `eventsPerPage`